### DISCUSSION TITLE
大神，请问一下能同时自定义多个直播源文件吗？若要同时处理，custom.json应该如何修改？

### DISCUSSION THREAD

**Discussion by lzedwin** [HerbertHe/iptv-sources#67] [Q&A]
例如，能否运行一次既可以同时对m_iptv、o_s_cn_cctv和q_bj_iptv_unicom等多个直播源文件进行自定义？

**Comment by HerbertHe** [comment]
多个上游目前可以分开编写规则，目前无法进行合并操作

**Comment by HerbertHe** [comment]
@lzedwin 因为rules本身就是一个数组，所以可以进行单独操作。但是目前不支持多源合并，其原因在于存在多个同名台重复的问题，没有好的解决方案，默认不支持生成时进行测速

**Reply by lzedwin** [reply]
感谢up主的详细讲解。在之前的视频中得知，现在是不支持多源合并操作的。但我现在遇到的问题是，每对一个源文件进行自定义必须写一个custom.json文件，运行一次yarn m3u后，再把针对其他源文件的custom.json覆盖之前的custom.json，然后再运行一次yarn m3u。假如我要自定义5个源文件，我就要重复5次上述步骤，这样比较繁琐。请问能否将5个rules数组写在同一个custom.json里面。我运行一次yarn m3u即可产生5个自定义后的源文件呢？这样5个rules分别针对5个不同的源文件，也生成不同的自定义源文件，不用担心多个同名台重复的问题，互不干扰。

**Reply by lzedwin** [reply]
或者能否让用户制作出的5个json文件（如custom1.json，custom2.json……custom5.json）可以同时挂载到/app/config下，只需要运行一次yarn m3u即可生成5个自定义后的源文件呢？

**Reply by HerbertHe** [reply]
#68 已经更新文档了，只要在 rules 下写五个规则就行了

**Reply by HerbertHe** [reply]
```json
{
  "rules": [
    { ...rule1 },
    { ...rule2 }
  ]
}
```

**Reply by HerbertHe** [reply]
多文件的形式不管是代码解析还是观感都非常的不雅，一个 custom.json 文件里本来就可以写多个 rule

**Reply by lzedwin** [reply]
感谢UP主的耐心解答和辛勤付出，刚才测试了一下，已经解决之前提及的问题。万分感谢

**Reply by newrookie001** [reply]
请问楼主，能否挂一份写好的custom.json文件，我参考一下怎么写。我规则部分upstream，exclude对应部分写好了，但下面的规则不会写了，尤其target部分，我目前docker中运行规则，显示任务执行成功，但看不到输出的结果
